### ISSUE TITLE
Training the CNN model of this study using your published dataset

### ISSUE THREAD

**Issue by Akihiko0123** [Boehringer-Ingelheim/anomaly-detection-in-histology#3]
Thank you very much for sharing your excellent work.

I would like to see if I can use your codes for my project.
Since we cannot prepare histological images stained by two methods, I am considering to use your kidney dataset to pretrain CNN model, then train my dataset by using one class svm.

I would appreciate it if you could let me know on the following.

・Is also the above use (cnn pretrain with your dataset and one class svm train with my dataset) included as the intended use ?
・Although your model seems to be MIT license, is not there any problem for me to use your dataset to train your CNN model? The output by using the pretrained CNN model + one class svm model (which will be trained by my dataset) may eventually be used commercially.

Thank you.

Best regards,
Akihiko

**Comment by igsing** [comment]
- You can directly use our already pre-trained CNN model for training one-class SVM on your own (kidney?) dataset.
- As far as I understand since our trained model is released under MIT license it allows commercial use.

**Comment by Akihiko-lpx** [comment]
I really appreciate for your helpful reply.

> ・You can directly use our already pre-trained CNN model for training one-class SVM on your own (kidney?) dataset.

Since your pre-trained CNN model shared in [here](https://osf.io/gqutd/) seems to be trained on liver not kidney, I thought I have to pre-train kidney dataset by myself for the next step one-class SVM.

Could you let me know if my understanding is correct that there is not big problem to train one-class SVM on my own kidney dataset using your pretrained model even if the model is trained with other organ like liver, as I may have misunderstood. 

or, I would appreciate it if you could tell me if we can find your already pretrained CNN model (kidney).

Thank you.

**Comment by Akihiko0123** [comment]
I really appreciate for your helpful reply.

>・You can directly use our already pre-trained CNN model for training one-class SVM on your own (kidney?) dataset.

Since your pre-trained CNN model shared in [here](https://osf.io/gqutd/) seems to be trained on liver not kidney, I thought I have to pre-train kidney dataset by myself for the next step one-class SVM.

Could you let me know if my understanding is correct that there is not big problem to train one-class SVM on my own kidney dataset using your pretrained model even if the model is trained with other organ like liver, as I may have misunderstood.

or, I would appreciate it if you could tell me if we can find your already pretrained CNN model (kidney).

Thank you.

**Comment by igsing** [comment]
The pre-trained CNN model was trained on histopathological images of DIFFERENT ORGANS, species, and stainings, as was described in the paper. This allows derived CNN representations to be used for various histopathological image analysis tasks (also for images coming from different organs). Our own one-class SVM classifier was trained on the liver tissue, but this SVM model you do not need, as you will train your target classifier on your kidney data.

**Comment by Akihiko0123** [comment]
Thank you so much for your prompt & kind explanation.
I am sorry but I have misunderstood on your pretrained CNN model.
I apologize for my persistence in asking,but is my current understanding correct that shared EfficientNet_B0_320_HE_Liver_Mouse_acc0.9762.pt and EfficientNet_B0_320_Masson_Liver_Mouse_acc0.9755.pt are trained on histopathological images of different organs, species?

I will read your paper more carefully.
Thank you again.

**Comment by igsing** [comment]
Yes, these models were trained on different organs and species, the first one on H&E, the second one on Masson Trichrome stained tissues (according to the names of the models). You can use one of these two models for your case of staining, Masson Trichrome or H&E correspondingly. 

The models also contain "liver" in their names, because these were trained with the center loss for the corresponding organ class. However, I think, the feature representations will also be good for other organs in the target task. Another option would be, indeed, to retrain the CNN model, again on all organs from our dataset, but with center loss that specifically minimizes the kidney class cluster size in the feature space. For such training you need to update 'centerloss_classes' in the configuration file, setting this variable either to 'kidney' (for Masson staining), or to 'he_kidney', for the case of H&E staining. I did not do experiments for image analysis/anomaly detection in kidney, so that I do not know if such retraining would essentially improve the results.

**Comment by Akihiko0123** [comment]
I really appreciate for your detailed explanation.

I understand I can use either model (H&E or Masson Trichrome) depending on the staining of my own patch images.
I also now understand that, as for the shared pre-trained model, the center loss was calculated based on the feature distance between the center of liver features and each liver patch feature, and the model is considered to be good also for the other organs regarding the target task.

We would first like to consider using your pre-trained model. 
Then, we might consider to retrain CNN model with updating 'centerloss_classes' in the configuration file to ''he_kidney' depending on the situation.

Thank you very much.
Best regards
